dot/types/body_extrinsics.go

```diff
@@ -86,73 +84,31 @@ func NewBodyFromExtrinsicStrings(ss []string) (*Body, error) {
 		exts = append(exts, b)
 	}
 
-	enc, err := scale.Encode(exts)
-	if err != nil {
-		return nil, err
-	}
-
-	body := Body(enc)
-	return &body, nil
+	return NewBody(BytesArrayToExtrinsics(exts)), nil
 }
 
-// AsExtrinsics decodes the body into an array of extrinsics
-func (b *Body) AsExtrinsics() ([]Extrinsic, error) {
-	exts := [][]byte{}
-
-	if len(*b) == 0 {
-		return []Extrinsic{}, nil
-	}
-
-	dec, err := scale.Decode(*b, exts)
+func (b *Body) AsSCALEEncodedBody() ([]byte, error) {
+	encodedBody, err := scale.Marshal(ExtrinsicsArrayToBytesArray(*b))
 	if err != nil {
 		return nil, err
 	}
 
-	return BytesArrayToExtrinsics(dec.([][]byte)), nil
+	return encodedBody, nil
 }
 
-// AsEncodedExtrinsics decodes the body into an array of SCALE encoded extrinsics
-func (b *Body) AsEncodedExtrinsics() ([]Extrinsic, error) {
-	exts := [][]byte{}
-
-	if len(*b) == 0 {
-		return []Extrinsic{}, nil
-	}
-
-	err := scale2.Unmarshal(*b, &exts)
-	if err != nil {
-		return nil, err
-	}
-
-	decodedExts := exts
-	ret := make([][]byte, len(decodedExts))
-
-	for i, ext := range decodedExts {
-		ret[i], err = scale2.Marshal(ext)
-		if err != nil {
-			return nil, err
-		}
-	}
+func (b *Body) DeepCopy() Body {
+	newExtrinsics := make([]Extrinsic, len([]Extrinsic(*b)))
+	copy(newExtrinsics, []Extrinsic(*b))
```

Comment:
I m not sure if this deep copies the extrinsics, as `Extrinsic` is `[]byte` which is essentially a pointer, so it's copying the pointers from one slice to another. I think you'll probably need a for loop to copy each extrinsic directly, but maybe double check?